embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/IngestionSmokeTest.java

```diff
@@ -408,6 +405,17 @@ private void waitForSegmentsToBeQueryable(int numSegments)
     );
   }
 
+  /**
+   * Gets the count of tasks with the given status for the specified datasource.
+   */
+  private int getTaskCount(String status, String dataSource)
```

Comment:
Consider moving this utility method to `EmbeddedClusterApis` itself so that it may be used by other tests too.